logger.js

```diff
@@ -10,135 +10,45 @@
+const path = require('path');
+const bunyan = require('bunyan');
 
-'use strict';
-
-var fs = require('fs');
-var util = require('util');
-var child_process = require('child_process');
-var path = require('path');
-var strftime = require('strftime').utc();
-
-require('colors');
-
-module.exports = function(config) {
-	config = config || {};
-	var exports = {};
-
-	config.levels = config.levels || {
-		none: 99,
-		trace: 0,
-		debug: 1,
-		log: 2,
-		info: 3,
-		warn: 4,
-		error: 5,
-		fatal: 6,
-	};
-
-	config.level_abbr = config.level_abbr || {
-		trace: 'trc',
-		debug: 'dbg',
-		log: 'log',
-		info: 'inf',
-		warn: 'WRN',
-		error: 'ERR',
-		fatal: 'FTL',
-	};
-
-	config.filename = `${process.cwd()}/${config.filename || 'logs.log'}`;
-
-	config.errorLevel = config.errorLevel || 'log';
-
-	child_process.execSync(`mkdir -p ${path.dirname(config.filename)}`);
-	var log_file = fs.createWriteStream(config.filename, { flags: 'a' });
-
-	exports.setLevel = function(errorLevel) {
-		config.errorLevel = errorLevel;
-	};
+const defaultLoggerConfig = {
+	filename: 'logs/lisk.log',
+	level: 'none',
+};
 
-	function snipFragileData(data) {
-		for (var key in data) {
-			if (key.search(/passphrase|password/i) > -1) {
-				data[key] = 'XXXXXXXXXX';
-			}
-		}
-		return data;
+const createDirIfNotExists = filename => {
+	if (!filename || filename === '') {
+		return;
 	}
+	const dir = path.dirname(filename);
+	if (!fs.existsSync(dir)) {
+		fs.mkdirSync(dir);
+	}
+};
 
-	Object.keys(config.levels).forEach(name => {
-		function log(message, data) {
-			var log = {
-				level: name,
-				timestamp: strftime('%F %T', new Date()),
-			};
-
-			if (message instanceof Error) {
-				log.message = message.stack;
-			} else {
-				log.message = message;
-			}
-
-			if (data && util.isObject(data)) {
-				log.data = JSON.stringify(snipFragileData(data));
-			} else {
-				log.data = data;
-			}
-
-			log.symbol = config.level_abbr[log.level]
-				? config.level_abbr[log.level]
-				: '???';
-
-			if (config.levels[config.errorLevel] <= config.levels[log.level]) {
-				if (log.data) {
-					log_file.write(
-						util.format(
-							'[%s] %s | %s - %s\n',
-							log.symbol,
-							log.timestamp,
-							log.message,
-							log.data
-						)
-					);
-				} else {
-					log_file.write(
-						util.format(
-							'[%s] %s | %s\n',
-							log.symbol,
-							log.timestamp,
-							log.message
-						)
-					);
-				}
-			}
-
-			if (
-				config.echo &&
-				config.levels[config.echo] <= config.levels[log.level]
-			) {
-				if (log.data) {
-					console.info(
-						`[${log.symbol.bgYellow.black}]`,
-						log.timestamp.grey,
-						'|',
-						log.message,
-						'-',
-						log.data
-					);
-				} else {
-					console.info(
-						`[${log.symbol.bgYellow.black}]`,
-						log.timestamp.grey,
-						'|',
-						log.message
-					);
-				}
-			}
-		}
-
-		exports[name] = log;
+const shouldShowSrc = level => level === 'debug' || level === 'trace';
```

Comment:
Not sure about `error` level but we should also show src on `fatal` level.

Reply:
i was thinking of cutting off by the log level. (debug or lower)
info or above is more for non-developer, so fatal should be already easily identifiable without src